### DISCUSSION TITLE
Import or conversion of xmind mindmaps to freeplane

### DISCUSSION THREAD

**Discussion by merlinuwe** [freeplane/freeplane#1874] [Help]
I have lots of xmind mindmaps. Most of them are linked together. How can I import or convert them for use in freeplane?

**Comment by macmarrum** [comment]
I don't know of any existing software that could convert .xmind to .mm files or import/export between these two applications.

The simplest solution might be to copy-paste the content from Xmind to Freeplane, though it would probably transfer only basic information.

Otherwise a converter would need to be created. And the task would require a bit of scripting knowledge, e.g. in Python or Groovy, etc. Here's some information of how it could be approached.

Today was the first time I installed and used Xmind to create a mind map. I can see the tool stores its data in a ZIP file (.xmind), which contains `content.json`.

I had created a library that allows to import JSON to a Freeplane mind map (as a new branch). The JSON data needs to be in a particular format, e.g.

```json
{
    "ID_585918462": {
        "@core": "Central Topic",
        "ID_339647333": {
            "@core": "Main Topic 1",
            "@details": "Sample details",
            "ID_282985455": {
                "@core": "Subtopic 1",
                "@link": "file:/home/m/tmp/Central%20Topic.xmind"
            },
            "ID_1384473417": {
                "@core": "Subtopic 2",
                "@link": "Central%20Topic.xmind"
            }
        },
        "ID_1998795443": {
            "@core": "Main Topic 2",
            "@note": "Sample note"
        },
        "ID_479731518": {
            "@core": "Main Topic 4"
        },
        "ID_1638775342": {
            "@core": "Main Topic 3"
        }
    }
}
```
_Note: `ID_...` can be any unique string_

I think much of Xmind textual node data could be converted, with a bit of effort, to the import-JSON format (above) and imported to Freeplane. Themes (aka Styles), colors, rich text (if any), images will most probably be lost in the process.

Here's the Import library and its Export counterpart:
- [Import.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/lib/io/github/macmarrum/freeplane/Import.groovy)
- [Export.groovy](https://raw.githubusercontent.com/macmarrum/freeplane-scripting/main/src/lib/io/github/macmarrum/freeplane/Export.groovy)

**Comment by jonadem** [comment]
Trying Import and Export.groovy scripts in Freeplane 1.13.3,  I get this
<img width="709" height="337" alt="image" src="https://github.com/user-attachments/assets/99fd7ae6-5e4d-47e3-ade5-674e75d9101b" />

**Reply by macmarrum** [reply]
@jonadem Could you please open a bug at https://github.com/macmarrum/freeplane-scripting/issues/new/choose

Please attach (as zipped files)
- your Freeplane log file from the session the error occurs (`Tools->Open user directory` → `logs` → `log.0`)
- the Groovy code/script that invokes Export/Import and causes the error
- the Import.groovy and Export.groovy files you used — they change over time, so this is to make sure you're using the latest version
